src/main/java/uk/gov/hmcts/probate/services/submit/services/SequenceService.java

```diff
@@ -1,37 +1,65 @@
 package uk.gov.hmcts.probate.services.submit.services;
 
+import java.util.Map;
+
 import com.fasterxml.jackson.databind.JsonNode;
 import com.fasterxml.jackson.databind.ObjectMapper;
 import com.fasterxml.jackson.databind.node.ObjectNode;
+import org.springframework.mail.javamail.JavaMailSenderImpl;
 import org.springframework.stereotype.Service;
 import org.springframework.beans.factory.annotation.Autowired;
 import uk.gov.hmcts.probate.services.submit.Registry;
 import uk.gov.hmcts.probate.services.submit.clients.PersistenceClient;
 
-import java.util.HashMap;
-import java.util.Map;
-
 @Service
 public class SequenceService {
     @Autowired
     Map<Integer, Registry> registryMap;
-
     @Autowired
     private PersistenceClient persistenceClient;
-    private ObjectMapper mapper = new ObjectMapper();
+    @Autowired
+    private JavaMailSenderImpl mailSender;
 
     private static int registryCounter = 1;
 
-    public synchronized JsonNode nextRegistryDataObject(String sequenceNumber) {
+    public synchronized JsonNode nextRegistry(long submissionReference) {
+        Registry nextRegistry = identifyNextRegistry();
+        return populateRegistrySubmitData(submissionReference, nextRegistry);
+    }
+
+    JsonNode populateRegistrySubmitData(long submissionReference, Registry registry) {
         ObjectMapper mapper = new ObjectMapper();
+        ObjectNode registryDataObject = mapper.createObjectNode();
         ObjectNode registryMapper = mapper.createObjectNode();
-        Registry nextRegistry = identifyNextRegistry();
-        registryMapper.put("submissionReference", sequenceNumber);
-        registryMapper.put("registryName", nextRegistry.capitalizeRegistryName());
-        registryMapper.put("registrySequenceNumber", Long.toString(getRegistrySequenceNumber(nextRegistry)));
-        registryMapper.put("registryEmail", nextRegistry.getEmail());
-        registryMapper.put("registryAddress", nextRegistry.getAddress());
-        return registryMapper;
+
+        registryDataObject.put("submissionReference", Long.toString(submissionReference));
+        registryMapper.put("name", registry.capitalizeRegistryName());
+        registryMapper.put("sequenceNumber", Long.toString(getRegistrySequenceNumber(registry)));
+        registryMapper.put("email", registry.getEmail());
+        registryMapper.put("address", registry.getAddress());
+        registryDataObject.set("registry", registryMapper);
+
+        return registryDataObject;
+    }
+
+    JsonNode populateRegistryResubmitData(long submissionReference, JsonNode formDataObject) {
+        ObjectMapper mapper = new ObjectMapper();
```

Comment:
You should be able to AutoWire in an ObjectMapper with Spring Boot, rather than creating a new instance each time, unless you really need to?